src/internal/connector/onedrive/item.go

```diff
@@ -45,31 +50,42 @@ func downloadItem(
 			}
 		}
 
-		if len(url) == 0 {
-			return nil, clues.New("extracting file url")
-		}
-
-		resp, err := ag.Get(ctx, url, nil)
+		rc, err = downloadFile(ctx, ag, url)
 		if err != nil {
-			return nil, clues.Wrap(err, "getting item")
+			return nil, err
```

Comment:
just so we get stack trace information
```suggestion
			return nil, clues.Stack(err)
```